src/Headers.ts

```diff
@@ -147,4 +155,15 @@ export default class HeadersPolyfill {
       }
     }
   }
+
+  /**
+   * Returns an array containing the values
+   * of all Set-Cookie headers associated
+   * with a response
+   */
+  getSetCookie(): string[] {
```

Comment:
Do you have some context around this method? It seems to be something new added, isn't it? I wonder how this will work on older versions of TypeScript. Would they scream "unexpected method "getSetCookie"" when validating `HeadersPolyfill` compliance with the standard `Headers` type?

Reply:
It should be fine to ignore it, given that we implement the rest of the contract. 

copied into the typescript playground, which looks to accept a HeadersPolyfill instance  in places it accepts a Headers instance, for both [5.1.6](https://www.typescriptlang.org/play?ts=5.1.6#code/C4TwDgpgBAEhCGATCAnAzgeQEYCsIGNgoBeKAJQIHsVEAeNYFASwDsBzAGigefagB9ujVmwDaAXQB8AKFCRYCZOgAyTBiSgBBFCnghaoniK5G+g02PFTp0-JRbqAchjIBZTcoCSALQCiAEQB9GF9Nf18yAGUALigAVxYmAEc46DQQAFssSgAbDUjM7JyACgByFmoM+BymAC8IRDgkVDRSgEpbe3UyTQB1YNDwskDHTVdfGPjElLTC3Py5ktLdAHcmpUd4DIhWjrsHIhCwiMCAKQxPR0Dwr1dPABUIjVKuUqh4NCh9hk6DqEjfPdAgBhDAYADSnl8A2Ow3BvgAmiMXO4vH5-M80BBgABaOyUADWTAgbw+Xy6wBsEAAHmBqERkAAzeBxHJEfA5D6fdYtAAKuRAjKYOTyAG9pFAoAB6KVQRyVap1BpQAAWilQUFFACIWFsIFrolr4FwsFqAL5CajwNgQAB0EqgYGYADd4MBoKJnG4PD4AjChpFxLEKHYaPRhOwTBG2JINKKzTZJTKoOCIBAwJ9gGqoFUwGARFAsNiVmmWFAs9BVqr1SgoLrtg7k-AWIhy9mKigqjV6q21c1a-XoMBKFAEFi29AcpRCXEwPbJU6mK73VBRD1+kchiMxhMg1BXPAwOHeJwhCfY6QWBAVvvD8UOg7vow4oRqMVWExgAB+WI89CeRIiEEP9MFwAggIUfs0FUBg2k1B0kwAKkQhDJUQqAAAEKhXYFm2wr41XwAkoCYRkJygAADD9gAokjPmbEiDmbfBoEoMiK0okCKNQqB0KLfAWXHLM3S+Tk0E+NQoHsHIQCgJlWGVVhyKwFBKBWLEUHnSU0KlVDSKgYoeNXUoQJeKATJrNB+RkoURVKcRbVYDk4mQNB30Ar9bSfFAX2HTTBwEQRSnaQKjOoxiGGYiA2MgpQrIFWyclQuDxW07Sn0Yz8mGqECNHCskQLCwDspyEDbUZahfHgfAVWKYpXRyVIuEHODiFjVK0u0rM1FtQ9IBbYpBy4BrUg6TqoDNLhurQMbtItCAcnHfTim0XQQEctBVr0dzPzaFKis-crKuq2rilEIaoBGiBxFa9qjMlaberzCABvu7ShreyUtvWtRvvq6pRqgL9LoBu0cEoVhik3E5zkua5fFuB4Ijg2Irre2a0rNDGJtHRboGW6j9vG7A8EIW0bWADAVhYXlVMgFBQE2bY3MJo6UCqmq6pakg7vGyUMquvLAPOvVxDex6+pexBDL5969Q4T6tB0PQNr+q64OBq7bXByHoeGWGrhuTw7keMgUZBxqIHRoysdQhNJQTB1RAKLJckc91dD88R73gtKUGxOIUDLR6XuEHZ7wdR20IJCAQDctpYk8D34CwHIICT1A3WoY8RF57SKtrYoMoCmKSfA20Y7j4pps9FEfXRf0IkDPbfc6kBiRyVtBztyPE14q748T5PU-T5O-Jz9g88lAuDIF0GpLIsuyYH6uVTUWvvTRP09ebom2471s0cx3uHUQ0PmHDhOoAz3QR5vrOUAMCwoxPKRW+n6hZ4pOs9QXqAl+ABXWObka5elRL6IIO8bp7zSu3BarYRbbCmmvNA5NsSDT1G0MWx8HZ9ylMhBC6EKDAEDg4d4lEABCIB3SRGjLRLEMwWAsT-tUPIHEB4sOrP2KAKxPwqnIWwJcL0f7bC0rxXSkoKYYO2LECwV8LACDrKyMUqF-YkKDm2deYD67b0GE3cQ50FTdggH+JmEBpEQCwYFJRIoT5IRQtpdCkRsT0TrNeC2qQoAzwYjSNQwACx9iUBFJgyByEUS4lJMChAuCfyQIgTM2ZAkan0p+OSlAdh1koEQao-skCyV8QwMRiEJHcHQYOWR0ZhqgwqSeK+zoIatg6vzb+HYuxKkQGYjQrTFT1FMXqCx2NQF1y3pAvRUQDHdOMR00WXSjFKj-AANVBv9S2gyUFrj6I3YYoxxiBltFiYAg05k9jMc1TBdjpQEMcVoZ6LZXFXhvILeww53hlgKf4vgSTaysDQCE6A8BOKWVopQKJwAYm1jiQk6AXySJkVSYgdJnx8I5MUPk6kfiiklMlgNcpZ5jAeIgDUkQdSGnv3JH8SZ7TOmXmOSYmsZiBmoTTkQf2aBcjOgaEsy2GhHoqg+EczsPSGhmOxpKYGFEAAkopHpSMpSczBk0oBSqumabi41Uagz7g9FB+yynyygKy9lnLQazSjpchxaEoD+AWtiDJAKYWMlUhkci4SgWRNJoAwhJTkDMvMbiuRqNSVNNhQZAAhLy-lLUYHaTUaQnuqFi60umdsWZgrjF9O2IytKPrbWaLQBvcBDcoGGLTVS0WqEa7ri2duXZDkc3ugFW0+V2xTV4KuZa4hpCoXuvAiwkU5E5XKi+agr1DpWH3mDFQMMz88WTzJbGjRQzN4QK2YGC5+CLW8XIAHIO3aQUer7Ww7MVZh2YodKsCd27Qx0BnRYKe5L1CrBApO69E9Tx3rjGaj+hcMqIIgFUy24g-4hxYGHeO0bJRPssqIStmy9Y1t3Ggw5LUgOkCPnNLVBqd1lig1Bdd7at2dt3eQ7IuQEBlkih8tgPC1QVghYCqCwLQXkuAPAH55CWIMzY2WL5Z7JR8rcv66MV9SNpwYsGhdwd1naJGauhyAmqY0zpqgUAjahUZr9Zg1tjYCPoXuLoDl6AMkcVdYxnt0TCFfFYQWDigiOVlgEiKLA1ViIzwQDVLhSg+NeOOjVWg9wUHaGo6QaakgZbpVYc5oisRwtpWmrEALv0UCcCMldIlkYjJCZPArcaYB4D+1A7EaayUeaXQaTlrqgXks-igIlzayWHTRpnkXFpv8lJLsLbo2EzcyWSmWh1nRozuvyY+Ip2mlB6aqajb1iLTmXNeXHdNILyCtHDJXcWwc4gzlILzaKiaPdcE6c3UQ7D9Eyz5bWix7jNnswD0szFVh-xsQ4mBNOIk0LLLvHEpQfA2V3SIEs7wrM5DWV0gcFbRxJSKbOOAK9wkxJL0WAkGSjKBy4fvb-CQIyA3ZPFoBECUEEIoTVvhEiGTECgNfmBsFVR2HSmw7e8SP8+ywA1EOfBg28NEam1bY7RkCRCBMHsGkyIlBtjdXYMUYdv5LIpUdgi0X4u16S4ebFPkCVhQlD2tIBXYvsTK7YINdxIF7xtCAA) and [5.2.2](https://www.typescriptlang.org/play?ts=5.2.2#code/C4TwDgpgBAEhCGATCAnAzgeQEYCsIGNgoBeKAJQIHsVEAeNYFASwDsBzAGigefagB9ujVmwDaAXQB8AKFCRYCZOgAyTBiSgBBFCnghaoniK5G+g02PFTp0-JRbqAchjIBZTcoCSALQCiAEQB9GF9Nf18yAGUALigAVxYmAEc46DQQAFssSgAbDUjM7JyACgByFmoM+BymAC8IRDgkVDRSgEpbe3UyTQB1YNDwskDHTVdfGPjElLTC3Py5ktLdAHcmpUd4DIhWjrsHIhCwiMCAKQxPR0Dwr1dPABUIjVKuUqh4NCh9hk6DqEjfPdAgBhDAYADSnl8A2Ow3BvgAmiMXO4vH5-M80BBgABaOyUADWTAgbw+Xy6wBsEAAHmBqERkAAzeBxHJEfA5D6fdYtAAKuRAjKYOTyAG9pFAoAB6KVQRyVap1BpQAAWilQUFFACIWFsIFrolr4FwsFqAL5CajwNgQAB0EqgYGYADd4MBoKJnG4PD4AjChpFxLEKHYaPRhOwTBG2JINKKzTZJTKoOCIBAwJ9gGqoFUwGARFAsNiVmmWFAs9BVqr1SgoLrtg7k-AWIhy9mKigqjV6q21c1a-XoMBKFAEFi29AcpRCXEwPbJU6mK73VBRD1+kchiMxhMg1BXPAwOHeJwhCfY6QWBAVvvD8UOg7vow4oRqMVWExgAB+WI89CeRIiEEP9MFwAggIUfs0FUBg2k1B0kwAKkQhDJUQqAAAEKhXYFm2wr41XwAkoCYRkJygAADD9gAokjPmbEiDmbfBoEoMiK0okCKNQqB0KLfAWXHLM3S+Tk0E+NQoHsHIQCgJlWGVVhyKwFBKBWLEUHnSU0KlVDSKgYoeNXUoQJeKATJrNB+RkoURVKcRbVYDk4mQNB30Ar9bSfFAX2HTTBwEQRSnaQKjOoxiGGYiA2MgpQrIFWyclQuDxW07Sn0Yz8mGqECNHCskQLCwDspyEDbUZahfHgfAVWKYpXRyVIuEHODiFjVK0u0rM1FtQ9IBbYpBy4BrUg6TqoDNLhurQMbtItCAcnHfTim0XQQEctBVr0dzPzaFKis-crKuq2rilEIaoBGiBxFa9qjMlaberzCABvu7ShreyUtvWtRvvq6pRqgL9LoBu0cEoVhik3E5zkua5fFuB4Ijg2Irre2a0rNDGJtHRboGW6j9vG7A8EIW0bWADAVhYXlVMgFBQE2bY3MJo6UCqmq6pakg7vGyUMquvLAPOvVxDex6+pexBDL5969Q4T6tB0PQNr+q64OBq7bXByHoeGWGrhuTw7keMgUZBxqIHRoysdQhNJQTB1RAKLJckc91dD88R73gtKUGxOIUDLR6XuEHZ7wdR20IJCAQDctpYk8D34CwHIICT1A3WoY8RF57SKtrYoMoCmKSfA20Y7j4pps9FEfXRf0IkDPbfc6kBiRyVtBztyPE14q748T5PU-T5O-Jz9g88lAuDIF0GpLIsuyYH6uVTUWvvTRP09ebom2471s0cx3uHUQ0PmHDhOoAz3QR5vrOUAMCwoxPKRW+n6hZ4pOs9QXqAl+ABXWObka5elRL6IIO8bp7zSu3BarYRbbCmmvNA5NsSDT1G0MWx8HZ9ylMhBC6EKDAEDg4d4lEABCIB3SRGjLRLEMwWAsT-tUPIHEB4sOrP2KAKxPwqnIWwJcL0f7bC0rxXSkoKYYO2LECwV8LACDrKyMUqF-YkKDm2deYD67b0GE3cQ50FTdggH+JmEBpEQCwYFJRIoT5IRQtpdCkRsT0TrNeC2qQoAzwYjSNQwACx9iUBFJgyByEUS4lJMChAuCfyQIgTM2ZAkan0p+OSlAdh1koEQao-skCyV8QwMRiEJHcHQYOWR0ZhqgwqSeK+zoIatg6vzb+HYuxKkQGYjQrTFT1FMXqCx2NQF1y3pAvRUQDHdOMR00WXSjFKj-AANVBv9S2gyUFrj6I3YYoxxiBltFiYAg05k9jMc1TBdjpQEMcVoZ6LZXFXhvILeww53hlgKf4vgSTaysDQCE6A8BOKWVopQKJwAYm1jiQk6AXySJkVSYgdJnx8I5MUPk6kfiiklMlgNcpZ5jAeIgDUkQdSGnv3JH8SZ7TOmXmOSYmsZiBmoTTkQf2aBcjOgaEsy2GhHoqg+EczsPSGhmOxpKYGFEAAkopHpSMpSczBk0oBSqumabi41Uagz7g9FB+yynyygKy9lnLQazSjpchxaEoD+AWtiDJAKYWMlUhkci4SgWRNJoAwhJTkDMvMbiuRqNSVNNhQZAAhLy-lLUYHaTUaQnuqFi60umdsWZgrjF9O2IytKPrbWaLQBvcBDcoGGLTVS0WqEa7ri2duXZDkc3ugFW0+V2xTV4KuZa4hpCoXuvAiwkU5E5XKi+agr1DpWH3mDFQMMz88WTzJbGjRQzN4QK2YGC5+CLW8XIAHIO3aQUer7Ww7MVZh2YodKsCd27Qx0BnRYKe5L1CrBApO69E9Tx3rjGaj+hcMqIIgFUy24g-4hxYGHeO0bJRPssqIStmy9Y1t3Ggw5LUgOkCPnNLVBqd1lig1Bdd7at2dt3eQ7IuQEBlkih8tgPC1QVghYCqCwLQXkuAPAH55CWIMzY2WL5Z7JR8rcv66MV9SNpwYsGhdwd1naJGauhyAmqY0zpqgUAjahUZr9Zg1tjYCPoXuLoDl6AMkcVdYxnt0TCFfFYQWDigiOVlgEiKLA1ViIzwQDVLhSg+NeOOjVWg9wUHaGo6QaakgZbpVYc5oisRwtpWmrEALv0UCcCMldIlkYjJCZPArcaYB4D+1A7EaayUeaXQaTlrqgXks-igIlzayWHTRpnkXFpv8lJLsLbo2EzcyWSmWh1nRozuvyY+Ip2mlB6aqajb1iLTmXNeXHdNILyCtHDJXcWwc4gzlILzaKiaPdcE6c3UQ7D9Eyz5bWix7jNnswD0szFVh-xsQ4mBNOIk0LLLvHEpQfA2V3SIEs7wrM5DWV0gcFbRxJSKbOOAK9wkxJL0WAkGSjKBy4fvb-CQIyA3ZPFoBECUEEIoTVvhEiGTECgNfmBsFVR2HSmw7e8SP8+ywA1EOfBg28NEam1bY7RkCRCBMHsGkyIlBtjdXYMUYdv5LIpUdgi0X4u16S4ebFPkCVhQlD2tIBXYvsTK7YINdxIF7xtCAA) (even down to [4.3.5](https://www.typescriptlang.org/play?ts=4.3.5#code/C4TwDgpgBAEhCGATCAnAzgeQEYCsIGNgoBeKAJQIHsVEAeNYFASwDsBzAGigefagB9ujVmwDaAXQB8AKFCRYCZOgAyTBiSgBBFCnghaoniK5G+g02PFTp0-JRbqAchjIBZTcoCSALQCiAEQB9GF9Nf18yAGUALigAVxYmAEc46DQQAFssSgAbDUjM7JyACgByFmoM+BymAC8IRDgkVDRSgEpbe3UyTQB1YNDwskDHTVdfGPjElLTC3Py5ktLdAHcmpUd4DIhWjrsHIhCwiMCAKQxPR0Dwr1dPABUIjVKuUqh4NCh9hk6DqEjfPdAgBhDAYADSnl8A2Ow3BvgAmiMXO4vH5-M80BBgABaOyUADWTAgbw+Xy6wBsEAAHmBqERkAAzeBxHJEfA5D6fdYtAAKuRAjKYOTyAG9pFAoAB6KVQRyVap1BpQAAWilQUFFACIWFsIFrolr4FwsFqAL5CajwNgQAB0EqgYGYADd4MBoKJnG4PD4AjChpFxLEKHYaPRhOwTBG2JINKKzTZJTKoOCIBAwJ9gGqoFUwGARFAsNiVmmWFAs9BVqr1SgoLrtg7k-AWIhy9mKigqjV6q21c1a-XoMBKFAEFi29AcpRCXEwPbJU6mK73VBRD1+kchiMxhMg1BXPAwOHeJwhCfY6QWBAVvvD8UOg7vow4oRqMVWExgAB+WI89CeRIiEEP9MFwAggIUfs0FUBg2k1B0kwAKkQhDJUQqAAAEKhXYFm2wr41XwAkoCYRkJygAADD9gAokjPmbEiDmbfBoEoMiK0okCKNQqB0KLfAWXHLM3S+Tk0E+NQoHsHIQCgJlWGVVhyKwFBKBWLEUHnSU0KlVDSKgYoeNXUoQJeKATJrNB+RkoURVKcRbVYDk4mQNB30Ar9bSfFAX2HTTBwEQRSnaQKjOoxiGGYiA2MgpQrIFWyclQuDxW07Sn0Yz8mGqECNHCskQLCwDspyEDbUZahfHgfAVWKYpXRyVIuEHODiFjVK0u0rM1FtQ9IBbYpBy4BrUg6TqoDNLhurQMbtItCAcnHfTim0XQQEctBVr0dzPzaFKis-crKuq2rilEIaoBGiBxFa9qjMlaberzCABvu7ShreyUtvWtRvvq6pRqgL9LoBu0cEoVhik3E5zkua5fFuB4Ijg2Irre2a0rNDGJtHRboGW6j9vG7A8EIW0bWADAVhYXlVMgFBQE2bY3MJo6UCqmq6pakg7vGyUMquvLAPOvVxDex6+pexBDL5969Q4T6tB0PQNr+q64OBq7bXByHoeGWGrhuTw7keMgUZBxqIHRoysdQhNJQTB1RAKLJckc91dD88R73gtKUGxOIUDLR6XuEHZ7wdR20IJCAQDctpYk8D34CwHIICT1A3WoY8RF57SKtrYoMoCmKSfA20Y7j4pps9FEfXRf0IkDPbfc6kBiRyVtBztyPE14q748T5PU-T5O-Jz9g88lAuDIF0GpLIsuyYH6uVTUWvvTRP09ebom2471s0cx3uHUQ0PmHDhOoAz3QR5vrOUAMCwoxPKRW+n6hZ4pOs9QXqAl+ABXWObka5elRL6IIO8bp7zSu3BarYRbbCmmvNA5NsSDT1G0MWx8HZ9ylMhBC6EKDAEDg4d4lEABCIB3SRGjLRLEMwWAsT-tUPIHEB4sOrP2KAKxPwqnIWwJcL0f7bC0rxXSkoKYYO2LECwV8LACDrKyMUqF-YkKDm2deYD67b0GE3cQ50FTdggH+JmEBpEQCwYFJRIoT5IRQtpdCkRsT0TrNeC2qQoAzwYjSNQwACx9iUBFJgyByEUS4lJMChAuCfyQIgTM2ZAkan0p+OSlAdh1koEQao-skCyV8QwMRiEJHcHQYOWR0ZhqgwqSeK+zoIatg6vzb+HYuxKkQGYjQrTFT1FMXqCx2NQF1y3pAvRUQDHdOMR00WXSjFKj-AANVBv9S2gyUFrj6I3YYoxxiBltFiYAg05k9jMc1TBdjpQEMcVoZ6LZXFXhvILeww53hlgKf4vgSTaysDQCE6A8BOKWVopQKJwAYm1jiQk6AXySJkVSYgdJnx8I5MUPk6kfiiklMlgNcpZ5jAeIgDUkQdSGnv3JH8SZ7TOmXmOSYmsZiBmoTTkQf2aBcjOgaEsy2GhHoqg+EczsPSGhmOxpKYGFEAAkopHpSMpSczBk0oBSqumabi41Uagz7g9FB+yynyygKy9lnLQazSjpchxaEoD+AWtiDJAKYWMlUhkci4SgWRNJoAwhJTkDMvMbiuRqNSVNNhQZAAhLy-lLUYHaTUaQnuqFi60umdsWZgrjF9O2IytKPrbWaLQBvcBDcoGGLTVS0WqEa7ri2duXZDkc3ugFW0+V2xTV4KuZa4hpCoXuvAiwkU5E5XKi+agr1DpWH3mDFQMMz88WTzJbGjRQzN4QK2YGC5+CLW8XIAHIO3aQUer7Ww7MVZh2YodKsCd27Qx0BnRYKe5L1CrBApO69E9Tx3rjGaj+hcMqIIgFUy24g-4hxYGHeO0bJRPssqIStmy9Y1t3Ggw5LUgOkCPnNLVBqd1lig1Bdd7at2dt3eQ7IuQEBlkih8tgPC1QVghYCqCwLQXkuAPAH55CWIMzY2WL5Z7JR8rcv66MV9SNpwYsGhdwd1naJGauhyAmqY0zpqgUAjahUZr9Zg1tjYCPoXuLoDl6AMkcVdYxnt0TCFfFYQWDigiOVlgEiKLA1ViIzwQDVLhSg+NeOOjVWg9wUHaGo6QaakgZbpVYc5oisRwtpWmrEALv0UCcCMldIlkYjJCZPArcaYB4D+1A7EaayUeaXQaTlrqgXks-igIlzayWHTRpnkXFpv8lJLsLbo2EzcyWSmWh1nRozuvyY+Ip2mlB6aqajb1iLTmXNeXHdNILyCtHDJXcWwc4gzlILzaKiaPdcE6c3UQ7D9Eyz5bWix7jNnswD0szFVh-xsQ4mBNOIk0LLLvHEpQfA2V3SIEs7wrM5DWV0gcFbRxJSKbOOAK9wkxJL0WAkGSjKBy4fvb-CQIyA3ZPFoBECUEEIoTVvhEiGTECgNfmBsFVR2HSmw7e8SP8+ywA1EOfBg28NEam1bY7RkCRCBMHsGkyIlBtjdXYMUYdv5LIpUdgi0X4u16S4ebFPkCVhQlD2tIBXYvsTK7YINdxIF7xtCAA)

Reply:
Just thought about testing that in the playground after responding with this. You are ahead, as always 👍 Thank you.